### PR TITLE
RavenDB-17239 stabilize rachis tests 

### PR DESCRIPTION
**Pull request by karmeli87**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-17239
https://issues.hibernatingrhinos.com/issue/RavenDB-17014
https://issues.hibernatingrhinos.com/issue/RavenDB-18635

### Additional description

- Make the test `RavenDB_13228` more stable, by injecting an entry to old leader and not rely on async task to do that before we disconnect the it. (RavenDB-17239)
- avoid returning `Task<Task>` since they aren't properly waited in the tests.
- Introduce `RavenTestHelper.AssertTrue` which will evaluate the message only upon failure.
- Handle a negotiation edge case that cause infinite negotiation loop (RavenDB-18635)
This happened when leader has a truncated log while the follower has a different appended term for the leader's first entry.

### Type of change

- Test fix
- Bug fix

### How risky is the change?

- Low 

### Backward compatibility

- Not relevant

### Is it platform specific issue?

- No

### Documentation update

- No documentation update is needed 

### Testing 

- Current test more stable when run locally
- New tests were added

### Is there any existing behavior change of other features due to this change?

- No

### UI work

- No UI work is needed

### CODE REVIEW

File: test/Tests.Infrastructure/RavenTestHelper.cs [hunk 1]
```diff
@@ -10,18 +11,16 @@
...
 using Sparrow.Platform;
 using Sparrow.Utils;
-using Xunit;
-using System.Text;
+using Voron.Platform.Posix;
 using Xunit.Sdk;
 using ExceptionAggregator = Raven.Server.Utils.ExceptionAggregator;
 
-namespace FastTests
+namespace Tests.Infrastructure
```
Comment (ppekrol) [inline test/Tests.Infrastructure/RavenTestHelper.cs]: revert

File: test/Tests.Infrastructure/RavenTestHelper.Assert.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Diagnostics;
+
+namespace Tests.Infrastructure
+{
+    public static partial class RavenTestHelper
+    {
+        public static class Assert
```
Comment (ppekrol) [inline test/Tests.Infrastructure/RavenTestHelper.Assert.cs]: unwrap this

**Review by ppekrol — COMMENTED** [review]
please revert all the usings as well
